aws-amplify/amplify-cli
# fix: removes userPool client lambda and update appclient secret

#### Description of changes
- removes UserPoolClientInputs custom resource and related custom resource (HostedUI, HostedUIprovider and OAuth) Custom resource depending on the userPoolClientLambda role.
- changed  logic to read appclient secret from cfn lambda to getting though aws-sdk call
- 


#### Issue #, if available



#### Description of how you validated changes
- unit tested
- Manually tested
- added a migration test and updated an e2e to check for specific use case
- https://app.circleci.com/pipelines/github/aws-amplify/amplify-cli/18414/workflows/0d17fbe2-7b5e-49ef-a6be-203af3e81a0b

#### Checklist



- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#tests)
- [ ] [Pull request labels](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#labels) are added

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
